plaso/parsers/azure_applicationgatewayaccess.py

```diff
@@ -0,0 +1,252 @@
+      first byte of a client request to be processed and its last-byte
+      sent in the response to the client. It's important to note that the
+      Time-Taken field usually includes the time that the request and
+      response packets are traveling over the network.
+    waf_evaluation_time (str): Length of time (in seconds) that it takes for
+      the request to be processed by the WAF.
+    waf_mode (str): Value can be either Detection or Prevention
+    transaction_id (str): Unique identifier to correlate the request received
+      from the client
+    ssl_enabled (str): Whether communication to the back-end pools used TLS.
+      Valid values are on and off.
+    ssl_cipher (str): Cipher suite being used for TLS communication
+      (if TLS is enabled).
+    ssl_protocol (str): SSL/TLS protocol being used (if TLS is enabled).
+    ssl_client_verify (str):
+    ssl_client_certificate_fingerprint (str):
+    ssl_client_certificate_issuer_name (str):
+    server_routed (str): The backend server that application gateway routes
+      the request to.
+    server_status (str): HTTP status code of the backend server.
+    server_response_latency (str): Latency of the response (in seconds)
+      from the backend server.
+    original_host (str): This field contains the original request host name
+    host (str): Address listed in the host header of the request. If rewritten
+      using header rewrite, this field contains the updated host name
+  """
+
+  DATA_TYPE = 'azure:applicationgatewayaccess:json'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(AzureApplicationGatewayAccessEventData, self).__init__(
+      data_type=self.DATA_TYPE
+    )
+
+    self.instance_id = None
+    self.client_ip = None
+    self.client_port = None
+    self.http_method = None
+    self.original_request_uri_with_args = None
+    self.request_uri = None
+    self.request_query = None
+    self.user_agent = None
+    self.http_status = None
+    self.http_version = None
+    self.received_bytes = None
+    self.sent_bytes = None
+    self.client_response_time = None
+    self.time_taken = None
+    self.waf_evaluation_time = None
+    self.waf_mode = None
+    self.transaction_id = None
+    self.ssl_enabled = None
+    self.ssl_cipher = None
+    self.ssl_protocol = None
+    self.ssl_client_verify = None
+    self.ssl_client_certificate_fingerprint = None
+    self.ssl_client_certificate_issuer_name = None
+    self.server_routed = None
+    self.server_status = None
+    self.server_response_latency = None
+    self.original_host = None
+    self.host = None
+
+
+class AzureApplicationGatewayAccessParser(interface.FileObjectParser):
+  """Parser for Azure application gateway access logs in JSON-L format."""
+
+  NAME = 'azure_applicationgatewayaccess'
+  DATA_FORMAT = 'Azure Application Gateway Access Logging'
+
+  _ENCODING = 'utf-8'
+
+  def _ParseAzureApplicationGatewayAccess(self, parser_mediator, file_object):
+    """
```

Comment:
style guide nit: Please start the description sentence right after the """
(reference: https://github.com/log2timeline/l2tdocs/blob/main/process/Style-guide.md#docstrings)